src/libstd/thread/mod.rs

```diff
@@ -112,6 +112,25 @@
 //! will want to make use of some form of **interior mutability** through the
 //! [`Cell`] or [`RefCell`] types.
 //!
+//! ## Naming threads
+//!
+//! Threads are able to have associated names for identification purposes. For example, the thread
+//! name is used in panic messages. By default, spawned threads are unnamed. To specify a name for
+//! a thread, build the thread with [`Builder`] and pass the desired thread name to
+//! [`Builder::name`]. To retrieve the thread name from within the thread, use [`Thread::name`].
```

Comment:
I'd note here that the name is also provided to the OS where applicable (e.g. pthread_setname_np).